Userland/DevTools/HackStudio/HackStudioWidget.h

```diff
@@ -138,6 +151,11 @@ class HackStudioWidget : public GUI::Widget {
 
     OwnPtr<Project> m_project;
 
+    Vector<ProjectLocation> m_locations_history;
+    Vector<ProjectLocation> m_locations_history_forward_list;
+    Optional<ProjectLocation> m_last_location;
+    bool m_locations_history_disabled { false };
```

Comment:
This feels like an unnecessarily difficult way to store the history.
Can we store it in a single vector and have a cursor index pointing at the current item?

Reply:
Yeah, this does simplify things. Done :+1: